pwrstat_prometheus.py

```diff
@@ -0,0 +1,94 @@
+"""Get output from pwrstat program and send results to REST or MQTT clients."""
+import re
+from typing import Dict
+
+from prometheus_client.core import GaugeMetricFamily, InfoMetricFamily
+
+import pwrstat_api
+
+
+class CustomCollector(object):
+    def __init__(self, labels: Dict[str, str] = None):
+        if labels is None:
+            labels = {}
+
+        self.label_names = list(labels.keys())
+        self.label_values = list(labels.values())
```

Comment:
This kind of static labeling goes against Prometheus best practices.

Reply:
I would disagree in that its not static. The labels provided here are from the running instance's config file, and allow for the specification of something like `rack: <identifier>`.  This allows people to specify meaningful labels to the instance, which is likely never to change.

It is also entirely optional.